packages/serverpod/lib/src/cache/local_cache.dart

```diff
@@ -94,10 +104,30 @@ class LocalCache extends Cache {
       }
     }
 
+    var pendingEntry = _inProgressCacheValues[key];
+    if (pendingEntry != null) {
+      return pendingEntry.then((v) => v as T);
+    }
+
     if (cacheMissHandler == null) return null;
 
-    var value = await cacheMissHandler.valueProvider();
-    if (value == null) return null;
+    T? value;
+    var completer = Completer<T?>();
+    try {
+      _inProgressCacheValues[key] = completer.future;
+
+      value = await cacheMissHandler.valueProvider();
+
+      completer.complete(value);
+
+      if (value == null) return null;
+    } catch (e, stackTrace) {
+      completer.completeError(e, stackTrace);
+
+      rethrow;
+    } finally {
+      unawaited(_inProgressCacheValues.remove(key));
```

Comment:
Question: Just for my understanding, we need this as the value stored is a future?

Reply:
yes, because we get a `Future?` from the map, the linter wants us to explicitly state that we don't want to handle it.

alternatively we could write `map[key] = null` to avoid this.